File: py/tests/money_test.py
class Dollar:
    def __init__(self, amount: int):
        self.amount = amount

    def times(self, multiple: int):
        return Dollar(self.amount * multiple)


def test_money():
    fiver = Dollar(5)
    tenner = fiver.times(2)
    assert tenner.amount == 10

File: ts/money.test.ts
import { expect, test } from 'vitest'

class Dollar {
  readonly amount: number

  constructor(amount: number) {
    this.amount = amount
  }

  times(i: number) {
    return new Dollar(this.amount * i)
  }

}

test('adds 1 + 2 to equal 3', () => {
  const fiver = new Dollar(5)
  const tenner = fiver.times(2)
  expect(tenner.amount).toBe(10)
})